test/accounts/coinbaseSmartWallet.ts

```diff
@@ -0,0 +1,129 @@
+import { encodeAbiParameters, encodeFunctionData, encodePacked, getAbiItem, getContract } from "viem";
+import { SignReturnType, sign } from "viem/accounts";
+import { AccountConfig, AccountFixtureReturnType } from "../benchmark";
+
+import hre from "hardhat";
+import { COINBASE_SMART_WALLET_ARTIFACTS } from "../artifacts/coinbaseSmartWallet";
+
+async function accountFixture(): Promise<AccountFixtureReturnType> {
+  const [walletClient] = await hre.viem.getWalletClients();
+  const publicClient = await hre.viem.getPublicClient();
+
+  for (const { address, bytecode } of Object.values(COINBASE_SMART_WALLET_ARTIFACTS)) {
+    await hre.network.provider.send("hardhat_setCode", [address, bytecode]);
+  }
+
+  const coinbaseSmartWalletFactory = getContract({
+    address: COINBASE_SMART_WALLET_ARTIFACTS.CoinbaseSmartWalletFactory.address,
+    abi: COINBASE_SMART_WALLET_ARTIFACTS.CoinbaseSmartWalletFactory.abi,
+    publicClient,
+    walletClient,
+  });
+
+  return {
+    createAccount: async (salt, ownerAddress) => {
+      const abiEncodedOwner = encodeAbiParameters([{ type: 'address' }], [ownerAddress])
+      return await coinbaseSmartWalletFactory.write.createAccount([
+        [abiEncodedOwner],
+        salt
+      ]);
+    },
+    getAccountAddress: async (salt, ownerAddress) => {
+      const abiEncodedOwner = encodeAbiParameters([{ type: 'address' }], [ownerAddress])
+      return await coinbaseSmartWalletFactory.read.getAddress([[abiEncodedOwner], salt]);
+    },
+    getOwnerSignature: async (owner, userOp, entryPoint) => {
+      const userOpHash = await entryPoint.read.getUserOpHash([userOp]);
+      // Key of owner account.
+      const privateKey =
+        "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+      const signature = await sign({ hash: userOpHash, privateKey });
```

Comment:
Could you do `owner.signMessage` instead? E.g. https://github.com/alchemyplatform/aa-benchmarks/blob/master/test/accounts/lightAccount.ts#L34-L36

Both methods work similarly well, but we get pings on "leaked keys" as a security issue in our repos, so would prefer to leave it out if possible

Reply:
@howydev signMessage has an EIP 191 prefix, which is a different thing and something we do not support.

Reply:
Safe does this too, this is fine. Separately, I did not realize viem exports this! Will have to go back and update https://github.com/alchemyplatform/aa-benchmarks/blob/aef6b6bfad2c5749cc7c2be5fcdeeec456a126d5/test/accounts/safe.ts#L269-L278

Reply:
Yeah I borrowed from the Safe file :)